src/bus/peer.c

```diff
@@ -794,10 +808,16 @@ int peer_queue_call(PolicySnapshot *sender_policy, NameSet *sender_names, ReplyO
 
         r = connection_queue(&receiver->connection, sender_user, message);
         if (r) {
-                if (CONNECTION_E_QUOTA)
-                        return PEER_E_QUOTA;
-                else
+                if (CONNECTION_E_QUOTA) {
+                        if (message->header->type == DBUS_MESSAGE_TYPE_SIGNAL &&
+                            peer_signal_is_solicited(receiver, sender_id, message)) {
+                                connection_shutdown(&receiver->connection);
```

Comment:
Maybe rename the function to `peer_message_is_solicited()` and check for signal-type there? Looks weird to have this type-check at a random place.